python/lsst/pipe/analysis/utils.py

```diff
@@ -41,11 +42,53 @@
            "writeParquet", "getRepoInfo", "findCcdKey", "getCcdNameRefList", "getDataExistsRefList",
            "orthogonalRegression", "distanceSquaredToPoly", "p1CoeffsFromP2x0y0", "p2p1CoeffsFromLinearFit",
            "lineFromP2Coeffs", "linesFromP2P1Coeffs", "makeEqnStr", "catColors", "setAliasMaps",
-           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac"]
+           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac",
+           "persistPlots"]
 
 
 NANOJANSKYS_PER_AB_FLUX = (0*units.ABmag).to_value(units.nJy)
 
+
+def persistPlots(plotList, dataId, butler):
+    """Persist plots and parse stats yielded by the supplied generator
+
+    Parameters
+    ----------
+    plotList : `list`
+        List of generators that will yield plots
+    dataId : `dict`
+        The dataId that will be used in persisting plots
+    butler : `lsst.daf.persistence.Butler`
+        The butler that might be used in persisting the plots
+    """
+    allStats = {}
+    allStatsHigh = {}
+    n = 0
+    for plots in plotList:
+        # In this conext next is called to "prime" the generator
+        # Each plotting function yields right after it is called so that
+        # each python execution frame is created, but not run. Calling next
+        # on the reference to the plot function begins the execution of the
+        # body, which will then yield each plot the function creates.
+        next(plots)
+        for plot in plots:
+            if hasattr(plot, 'description'):
```

Comment:
Seems to assume that it can have either a description or filename but not both. I'd either double check that it doesn't have both or check for the filename put the filename check first. 
```
if hasattr(plot, 'filename'):
        key = plot.filename
        plot.fig.savefig(plot.filename, dpi=120)
```

Reply:
In practice this will never (currently) have both, and is an artifact of how lauren and sophie are each using a different way to output plots. It is not bad to be defensive for the future and check if there is both, but I am not sure what it does to move the filename first, other than potentially triggering an exception in the butler if a file already exists (which maybe is what you had in mind). I had somewhat had in mind supporting both in the struct so that in the future there could be an output through the butter and an optional output to a file (maybe something specified by the user in their home directory, or webserver or something) through config. But maybe it is premature to try supporting that, or even to support that at all.

Reply:
What do we want it to do if it has both? I think that the filename (generated through a butler.get("stuff_filename", dataId) where the dataId includes a description) will go away as part of coalescing the butler calls into only one place. Then if we are keeping it so that people can save things in their own places do we want that to be done preferentially when the filename is specified?

Reply:
Filename is generated by using the butler so it had to go anyway.